dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/relationship/DefaultRelationshipService.java

```diff
@@ -42,23 +43,47 @@
 import org.hisp.dhis.relationship.RelationshipItem;
 import org.hisp.dhis.relationship.RelationshipType;
 import org.hisp.dhis.trackedentity.TrackedEntity;
+import org.hisp.dhis.tracker.TrackerType;
 import org.hisp.dhis.tracker.acl.TrackerAccessManager;
 import org.hisp.dhis.tracker.export.Page;
 import org.hisp.dhis.tracker.export.PageParams;
 import org.hisp.dhis.user.CurrentUserUtil;
 import org.hisp.dhis.user.UserDetails;
+import org.mapstruct.factory.Mappers;
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
 
 @Service("org.hisp.dhis.tracker.export.relationship.RelationshipService")
 @Transactional(readOnly = true)
 @RequiredArgsConstructor
 public class DefaultRelationshipService implements RelationshipService {
-
+  private static final RelationshipItemMapper RELATIONSHIP_ITEM_MAPPER =
+      Mappers.getMapper(RelationshipItemMapper.class);
   private final TrackerAccessManager trackerAccessManager;
   private final RelationshipStore relationshipStore;
   private final RelationshipOperationParamsMapper mapper;
 
+  // TODO(DHIS2-18883) Pass fields params as a parameter
+  @Override
+  public Set<RelationshipItem> getRelationshipItems(TrackerType trackerType, UID uid) {
+    List<RelationshipItem> relationshipItems =
+        switch (trackerType) {
+          case TRACKED_ENTITY -> relationshipStore.getRelationshipItemsByTrackedEntity(uid);
+          case ENROLLMENT -> relationshipStore.getRelationshipItemsByEnrollment(uid);
+          case EVENT -> relationshipStore.getRelationshipItemsByEvent(uid);
+          case RELATIONSHIP -> throw new IllegalArgumentException("Unsupported type");
+        };
+    return relationshipItems.stream()
+        .filter(
+            ri ->
+                trackerAccessManager
+                    .canRead(CurrentUserUtil.getCurrentUserDetails(), ri.getRelationship())
```

Comment:
Is it still an open question if we should show a relationship with only one of the items in case the user does not have access to the other?

Reply:
It is temporary closed :grimacing: 
It is something that in the future could make sense. For now we agreed that if the user has no access to one of the items then he has no access to the relationship

Reply:
I think so. But we should get back to this in the ACL big picture. 
Since we do not have anyway of showing "just a relationship, without what it relates to" in the Capture App, we can keep it this way. Not show the the relationship if no access to relationshipitem.